default/src/components/signer.js

```diff
@@ -126,126 +147,279 @@ const GetTxParams = () => {
   );
 };
 
-// NOTE: Update the types in this function once Algosdk will support BigInt
+/* eslint-disable sonarjs/cognitive-complexity */
 const SendASA = () => {
-  let asaData = YAMLData.default.default.asa;
+  let goldAsaInfo = Gold.default.default.asa;
+  let teslaAsaInfo = Tesla.default.default.asa;
   let deployedASANames = [];
-  let deployedASAIds = [];
-  let asaId;
-  for (const [key, value] of Object.entries(asaData)) {
-    deployedASANames.push(key);
-    deployedASAIds.push(value.assetIndex);
+  let asaInfo = {};
+  const assets = Object.entries({ ...goldAsaInfo, ...teslaAsaInfo });
+  for (const [name, info] of assets) {
+    asaInfo[name] = info;
```

Comment:
can we use `assets` above instead of asaInfo ?

Reply:
No, assets is not a map, so had to construct one.
![image](https://user-images.githubusercontent.com/33264364/121691405-95d43a00-cae4-11eb-9f40-74aeaa3a82c2.png)